test/plugins/sails.spec.js

```diff
@@ -0,0 +1,114 @@
+'use strict'
+
+const agent = require('./agent')
+const plugin = require('../../src/plugins/sails')
+
+wrapIt()
+
+describe('Plugin', () => {
+  let tracer
+  let Sails
+
+  describe('sails', () => {
+    withVersions(plugin, 'sails', version => {
+      beforeEach(() => {
+        tracer = require('../..')
+      })
+
+      before(() => {
+        return agent.load(plugin, 'sails').then(() => {
+          Sails = require(`../../versions/sails@${version}`).get()
+        })
+      })
+
+      after(() => {
+        Sails.lower()
+
+        agent.close()
+      })
+
+      it('traces registered classic actions', done => {
+        Sails.registerAction(() => { return 'test response' }, 'testAction')
+
+        agent.use(traces => {
+          expect(traces[0][0]).to.have.property('name', 'action testAction')
+          expect(traces[0][0]).to.have.property('service', 'test')
+          expect(traces[0][0].meta).to.have.property('component', 'Sails.js')
+          expect(traces[0][0].meta).to.have.property('span.kind', 'server')
+          expect(traces[0][0].meta).to.have.property('classicAction', 'true')
+        }).then(done)
+          .catch(done)
+
+        expect(Sails.getActions()['testaction']()).to.eq('test response')
+      })
+
+      it('traces registered actions2 actions', done => {
+        function action2 () {
+          return 'action2 response'
+        }
+        // clumsy way of faking the object that machine-as-action returns
+        action2.IS_MACHINE_AS_ACTION = true
```

Comment:
Not sure I understand this, or why registering something like `{fn: action2}` isn't adequate?  Definitely don't know much about sails though, so if this is an internal constant a link would be helpful.

Reply:
By the time it reaches this method it is no longer the dictionary I think you are thinking of. It gets processed though `machine-as-action`, so this is the easiest way to produce that object without actually running it through that module. The name of member is meaningless.

I can link to the module code if that's useful, but I'm not sure what else would be helpful.

Reply:
I think I understand now. This is my fault, I'll fix the poor implementation I've done here.

Reply:
There should now be a more clear distinction between the regular and middleware registration paths for controllers and this path of registering action2 dictionaries programmatically will actually work